test/internal/request/RequestQueue.get.spec.js

```diff
@@ -238,75 +244,175 @@ describe('#get', function() {
+        const queue = new RequestQueue(model, scheduler);
+
+        let zip;
+        zip = zipSpy(2, () => {
+            expect(queue._requests.length).toBe(0);
+            expect(zip).toHaveBeenCalledTimes(2);
+
+            expect(dsGetSpy).toHaveBeenCalledTimes(2);
+
+            // Paths should still be collapsed
+            expect(dsGetSpy).toHaveBeenNthCalledWith(1, expect.anything(), [["videos", { from: 0, to: 1 }, "title"]]);
+            expect(dsGetSpy).toHaveBeenNthCalledWith(2, expect.anything(), [["videos", { from: 1, to: 2 }, "title"]]);
+
+            done();
+        });
+
+        queue.get([videos0, videos1], [videos0, videos1], zip);
+        queue.get([videos1, videos2], [videos1, videos2], zip);
+    });
+
+    it("does not collapse paths when it is disabled", done => {
+        const scheduler = new ImmediateScheduler();
+        const dsGetSpy = jest.fn();
+        const source = new LocalDataSource(Cache(), { wait: 100, onGet: dsGetSpy });
+        const model = new Model({ source, disablePathCollapse: true });
+        const queue = new RequestQueue(model, scheduler);
+
+        let zip;
+        zip = zipSpy(2, () => {
+            expect(queue._requests.length).toBe(0);
+            expect(zip).toHaveBeenCalledTimes(2);
+
+            // Requests should still be deduplicated
+            expect(dsGetSpy).toHaveBeenCalledTimes(2);
+            expect(dsGetSpy).toHaveBeenNthCalledWith(1, expect.anything(), [videos0, videos1]);
+            expect(dsGetSpy).toHaveBeenNthCalledWith(2, expect.anything(), [videos2]);
+
+            done();
+        });
+
+        queue.get([videos0, videos1], [videos0, videos1], zip);
+        queue.get([videos1, videos2], [videos1, videos2], zip);
+    });
+
+    it("does not collapse paths or dedupe requests when disabled", done => {
+        const scheduler = new ImmediateScheduler();
+        const dsGetSpy = jest.fn();
+        const source = new LocalDataSource(Cache(), { wait: 100, onGet: dsGetSpy });
+        const model = new Model({ source, disablePathCollapse: true, disableRequestDeduplication: true });
+        const queue = new RequestQueue(model, scheduler);
+
+        let zip;
+        zip = zipSpy(2, () => {
+            expect(queue._requests.length).toBe(0);
+            expect(zip).toHaveBeenCalledTimes(2);
+
+            // No path collapse, no request dedupe
+            expect(dsGetSpy).toHaveBeenCalledTimes(2);
+            expect(dsGetSpy).toHaveBeenNthCalledWith(1, expect.anything(), [videos0, videos1]);
+            expect(dsGetSpy).toHaveBeenNthCalledWith(2, expect.anything(), [videos1, videos2]);
+
+            done();
+        });
+
+        queue.get([videos0, videos1], [videos0, videos1], zip);
+        queue.get([videos1, videos2], [videos1, videos2], zip);
+    });
+
+    it("combines batched paths without collapse when it is disables", done => {
+        const scheduler = new ASAPScheduler();
```

Comment:
what's the relevance, if any, of `ImmediateScheduler` vs `ASAPScheduler`?

Reply:
- ImmediateScheduler: executes `flushGetRequest` synchronously, never batches (because they'll be marked 'sent' directly after the respective `model.get()` call)
- ASAPScheduler: schedules fetches on the next tick, successive `model.get()` calls are batched